### PR TITLE
Fixed issue #12

### PR DESCRIPTION
**Pull request by nofaralfasi**
[Issue](https://github.com/nofaralfasi/todo_list/issues/12)
Changed task.due_date type to datetime, and task.priority type to integer.

### CODE REVIEW

File: app/views/tasks/_form.html.erb [hunk 1]
```diff
@@ -6,13 +6,13 @@
 </div>
 
 <div class="form-group">
-  <%= f.label :due_date %>
-  <%= f.text_field :due_date, class: "form-control", placeholder: "Enter due date" %>
+  <%= f.label :due_date %><br/>
+  <%= f.datetime_select(:due_date, { ampm: true, order: [:day, :month, :year], start_year: Date.today.year }, { class: 'form-control date-select' }) %>
```
Comment (stejskalleos) [inline app/views/tasks/_form.html.erb:10]: When I try to create new task I got error:
```ruby
ActionView::Template::Error (undefined method `min' for "{3=>11, 2=>5, 1=>2022, 4=>10, 5=>51}":String):
     7: 
     8: <div class="form-group">
     9:   <%= f.label :due_date %><br/>
    10:   <%= f.datetime_select(:due_date, { ampm: true, order: [:day, :month, :year], start_year: Date.today.year }, { class: 'form-control date-select' }) %>
    11: </div>
    12: 
    13: <div class="form-group">
```

Full development.log:
```
Started POST "/lists/2/tasks" for 127.0.0.1 at 2022-05-11 12:52:30 +0200
Processing by TasksController#create as HTML
  Parameters: {"authenticity_token"=>"[FILTERED]", "task"=>{"title"=>"", "due_date(3i)"=>"11", "due_date(2i)"=>"5", "due_date(1i)"=>"2022", "due_date(4i)"=>"10", "due_date(5i)"=>"51", "priority"=>"1", "status"=>"false", "label_id"=>""}, "commit"=>"Create Task", "list_id"=>"2"}
  List Load (0.2ms)  SELECT "lists".* FROM "lists" WHERE "lists"."id" = $1 LIMIT $2  [["id", 2], ["LIMIT", 1]]
  ↳ app/controllers/tasks_controller.rb:53:in `get_list'
  Rendering layout layouts/application.html.erb
  Rendering tasks/new.html.erb within layouts/application
  Rendered application/_error_messages.html.erb (Duration: 1.5ms | Allocations: 976)
  Rendered tasks/_form.html.erb (Duration: 5.7ms | Allocations: 5401)
  Rendered tasks/new.html.erb within layouts/application (Duration: 6.1ms | Allocations: 5557)
  Rendered layout layouts/application.html.erb (Duration: 6.2ms | Allocations: 5631)
Completed 500 Internal Server Error in 10ms (ActiveRecord: 0.2ms | Allocations: 7515)


  
ActionView::Template::Error (undefined method `min' for "{3=>11, 2=>5, 1=>2022, 4=>10, 5=>51}":String):
     7: 
     8: <div class="form-group">
     9:   <%= f.label :due_date %><br/>
    10:   <%= f.datetime_select(:due_date, { ampm: true, order: [:day, :month, :year], start_year: Date.today.year }, { class: 'form-control date-select' }) %>
    11: </div>
    12: 
    13: <div class="form-group">
  
app/views/tasks/_form.html.erb:10
app/views/tasks/new.html.erb:6
app/views/tasks/new.html.erb:4
app/controllers/tasks_controller.rb:17:in `create'
```

**Review by stejskalleos — CHANGES_REQUESTED** [review]

File: app/views/tasks/_form.html.erb (hunk 1 above)
Comment (nofaralfasi) [reply app/views/tasks/_form.html.erb:10]: I am trying to figure out why you are getting this error, cause for me it works fine.
Maybe it is related to the changes I made on the `tasks` table?
Did you try migrate back down and then running the `rails db:migrate` command before running the server?

Comment (stejskalleos) [reply app/views/tasks/_form.html.erb:10]: Yop, db cleanup did the job ...

**Review by stejskalleos — APPROVED** [review]
:heavy_check_mark: **LGTM**

**Pull request merged**